src/test/java/io/reactivex/flowable/FlowableTests.java

```diff
@@ -1060,4 +1060,19 @@ public void toObservableRange() {
     public void toObservableError() {
         Flowable.error(new TestException()).toObservable().test().assertFailure(TestException.class);
     }
+
+    @Test
+    public void zipIterableObject() {
+        final List<Flowable<Integer>> flowables = Arrays.asList(Flowable.just(1, 2, 3), Flowable.just(1, 2, 3));
+        Flowable.zip(flowables, new Function<Object[], Object>() {
```

Comment:
Yeah, this always passes because it is compatible with both `? super T[]` and `? super Object[]`. I have no idea how to test with `Function<Integer[], Object>` because it doesn't even compile with the latter signature. Leave this as is.